src/modules/social-feed/comment/comment.service.ts

```diff
@@ -0,0 +1,77 @@
+import { Comment } from "@prisma/client";
+import { CreateCommentInput } from "src/core/dto/social-feed.dto.js";
+import { Service } from "typedi";
+
+import { prisma } from "../../../prisma/index.js";
+
+@Service()
+export default class CommentService {
+  async createComment(
+    commentInput: CreateCommentInput,
+    userId: number
+  ): Promise<Comment> {
+    try {
+      const comment = await prisma.comment.create({
+        data: {
+          userId: userId,
+          postId: commentInput.postId,
+          content: commentInput.content,
+        },
+      });
+      return comment;
+    } catch (error) {
+      throw new Error((error as Error).message);
+    }
+  }
+  async deleteComment(commentId: number, userId: number): Promise<boolean> {
+    try {
+      const comment = await prisma.comment.findUnique({
+        where: { id: commentId },
+      });
+      if (comment.userId !== userId) {
```

Comment:
when I run this I get an error that comment is possibly null, needs type narrowing check.